nupic/research/frameworks/tensorflow/constraints/sparse_weights.py

```diff
@@ -0,0 +1,72 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2018, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import tensorflow as tf
+from tensorflow.python import keras
+
+
+class SparseWeights(keras.constraints.Constraint):
+    """
+    Sparse weights constraint.
+
+    Constrains the weights to a fixed sparsity rate where a fixed number of
+    weights are always zeros.
+
+    :param sparsity:
+      Percentage of weights that are allowed to be non-zero. Default 0.5
+    :type sparsity: float
+    """
+
+    def __init__(self, sparsity=0.5, name=None):
```

Comment:
Should we call this percent_on instead of sparsity?

Reply:
Good point, should we also rename the pytorch version?
https://github.com/numenta/nupic.torch/blob/master/nupic/torch/modules/sparse_weights.py#L105

Reply:
This is fundamentally different than the percent_on arg in KWinners right? If so, I see some value in using unambiguous terms.

Reply:
Yes, whatever term we use, we should make the usage clear. We could use percent_non_zero if that is clearer.

I've found that the word sparsity is really tough for people once we get to specifics. Does sparsity=0.2 mean 20% are non-zero or that 80% are non-zero?

You could argue the current usage is incorrect:
`        :param weight_sparsity:
          Pct of weights that are allowed to be non-zero in the layer.
`

Reply:
I will rename `sparsity ` to `percent_non_zero` on `nupic.research` tensorflow version only for now. 
The `nupic.torch` change is a bigger change involving multiple repositories. I will do that next week.